switch_mod/project/unitcommit/commit.py

```diff
@@ -249,6 +255,72 @@ def define_components(mod):
             if t == t2))
     mod.cost_components_tp.append('Total_Startup_OM_Costs')
 
+    mod.g_min_uptime = Param(
+        mod.GENERATION_TECHNOLOGIES,
+        within=NonNegativeReals,
+        default=0.0)
+    mod.g_min_downtime = Param(
+        mod.GENERATION_TECHNOLOGIES,
+        within=NonNegativeReals,
+        default=0.0)
+    # timepoints when projects with uptime or downtime constraints are active
+    mod.PROJ_MIN_UPTIME_DISPATCH_POINTS = Set(dimen=2, initialize=lambda m: [
+        (pr, tp) 
+            for pr in m.PROJECTS if m.g_min_uptime[m.proj_gen_tech[pr]] > 0.0
+                for tp in m.PROJ_ACTIVE_TIMEPOINTS[pr] 
+    ])
+    mod.PROJ_MIN_DOWNTIME_DISPATCH_POINTS = Set(dimen=2, initialize=lambda m: [
+        (pr, tp) 
+            for pr in m.PROJECTS if m.g_min_downtime[m.proj_gen_tech[pr]] > 0.0
+                for tp in m.PROJ_ACTIVE_TIMEPOINTS[pr] 
+    ])
+    
+    def tp_prev(m, tp, n=1):
+        # find nth previous timepoint, wrapping from start to end of day
+        return m.TS_TPS[m.tp_ts[tp]].prevw(tp, n)
+    # min_time_projects = set()
+    def min_time_rule(m, pr, tp, up):
+        # if (pr, up) not in min_time_projects:
+        #     min_time_projects.add((pr, up))
+        #     print "creating min {}time constraint for {}".format('up' if up else 'down', pr)
+        # note: this enforces a simple rule: all capacity turned on in the last x hours must 
```

Comment:
Move notes like this into the documentation so it can be available everywhere, not just in the code. If you also want notes like this in the function itself, consider using a docstring so it can be extracted during the auto-documentation process. Ex:
```python
    def min_time_rule(m, pr, tp, up):
        """
        This enforces a simple rule: all capacity turned on ...
        """
        ...
```